### PR TITLE
Add validation initialState - Invalid

### LINKED ISSUE #2940: Add schema validation support to system component on provided options
### Expected behavior
Provided options for system component need validation first.

### Actual behavior

No validation.

### Which version(s) does this affect? (Environment, OS, etc...)

1.6.0

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
Limited validation initial state.

### Solution
More rigid validation schema added for new app state.
Will be replaced by branch `feature/improve_overall_configuration`.

### How to test it?
Start app without schema validation errors.

### Review checklist

* The PR resolves
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/controller/schema/application.js [hunk 1]
```diff
@@ -211,24 +211,46 @@ module.exports = {
...
 					protocolVersion: {
 						type: 'string',
+						example: 1,
+						format: 'protocolVersion',
+						description:
+							'The protocol version of Lisk Core that the peer node runs on.',
 					},
 					nonce: {
 						type: 'string',
+						example: 'sYHEDBKcScaAAAYg',
```
Comment (shuse2) [inline framework/src/controller/schema/application.js]: I think nonce has to be specific length atm

File: framework/src/controller/schema/application.js [hunk 2]
```diff
@@ -211,24 +211,46 @@ module.exports = {
 				properties: {
 					nethash: {
 						type: 'string',
+						maxLength: 64,
```
Comment (shuse2) [inline framework/src/controller/schema/application.js]: nethash has to be specific length with hex

File: framework/src/controller/schema/application.js (hunk 1 above)
Comment (michielmulders) [reply framework/src/controller/schema/application.js]: 16 I just found

File: framework/src/controller/schema/application.js [hunk 3]
```diff
@@ -211,24 +211,47 @@ module.exports = {
 				properties: {
 					nethash: {
 						type: 'string',
+						format: 'hex',
```
Comment (diego-G) [inline framework/src/controller/schema/application.js]: I'm not sure about this... Why `hex` @shuse2? What you can add here is the length `64`.

Comment (michielmulders) [reply framework/src/controller/schema/application.js]: `hex` formats for /^[a-f0-9]+$/`, I thought it also checked for length `64` but apparently it doesn't.

**Comment by nazarhussain** [comment]
@michielmulders This PR seems ok, but I will hold it for a bit to first merge the feature branch to development. This is small PR so a bit easy to resolve conflicts if there is any.

File: framework/src/controller/schema/application.js [hunk 4]
```diff
@@ -173,24 +173,50 @@ module.exports = {
 						properties: {
 							nethash: {
 								type: 'string',
+								format: 'hex',
+								minLength: 64,
+								maxLength: 64,
 							},
 							version: {
 								type: 'string',
+								example: 'v0.8.0',
```
Comment (nazarhussain) [inline framework/src/controller/schema/application.js]:
```suggestion
								example: '1.6.0-rc.3',
```

Comment (nazarhussain) [reply framework/src/controller/schema/application.js]: We don't use any prefixes like `v` for versions.

**Pull request closed without merging**